File: order-service/src/main/java/com/ecommerce/orderservice/kafka/config/KafkaTopic.java
package com.ecommerce.orderservice.kafka.config;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum KafkaTopic {

    // Produce
    ORDER_CREATED(TopicName.ORDER_CREATED),
    ORDER_FAILED(TopicName.ORDER_FAILED),
    ORDER_CANCELLED(TopicName.ORDER_CANCELLED),

    // Consume
    INVENTORY_FAILED(TopicName.INVENTORY_FAILED),
    PAYMENT_SUCCESS(TopicName.PAYMENT_SUCCESS),
    PAYMENT_FAILED(TopicName.PAYMENT_FAILED),
    ;

    private final String topicName;

    public static final class TopicName {

        // Produce
        public static final String ORDER_CREATED = "order.created";
        public static final String ORDER_FAILED = "order.failed";
        public static final String ORDER_CANCELLED = "order.cancelled";

        // Consume
        public static final String INVENTORY_FAILED = "inventory.failed";
        public static final String PAYMENT_SUCCESS = "payment.success";
        public static final String PAYMENT_FAILED = "payment.failed";

        private TopicName() {
        }
    }
}

File: review-service/src/main/kotlin/com/ecommerce/reviewservice/kafka/config/KafkaTopic.kt
package com.ecommerce.reviewservice.kafka.config

enum class KafkaTopic(val topicName: String) {

    REVIEW_CREATED(TopicName.REVIEW_CREATED)
    ;

    // object는 선언과 동시에 인스턴스 딱 하나 만드는 문법
    object TopicName {

        /* const는 컴파일 타임에 값이 확정된 상수, val은 한 번 할당 후 변경 불가 지만 런타임값
        * const는 object, companion object안쪽 혹은 top-level에서 사용 가능
         */
        const val REVIEW_CREATED = "review.created"
    }
}